### DISCUSSION TITLE
Error when trying to setup Solana Jupiter bot

### DISCUSSION THREAD

**Discussion by searcher-py** [ARBProtocol/solana-jupiter-bot#19] [Q&A]
Hello Patel how are you?

Could you help me out how to overcome YARN version problems on UBUNTU 22.04 server when trying to setup Solana Jupiter Bot after cloning the repo.

I have installed previously

Curl
Yarn
Jupiter

Error:
ubuntu@ubuntuserver2204:~/solana-jupiter-bot$ yarn
yarn install v1.22.19
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/5] Validating package.json...
error solana-jupiter-bot@0.0.10-beta: The engine "node" is incompatible with this module. Expected version ">=16". Got "12.22.9"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

Do you have a preferred Lnux Dist. not to run into these kind of errors?

Thank you very much for your time
Looking forward to continue reviewing and trying your Bot

Greetings
Klaus
klaus.hilbrandt@gmail.com

**Comment by pmioduszewski** [comment]
Hi,
`The engine "node" is incompatible with this module. Expected version ">=16". Got "12.22.9"` - try to run it on the [Node 16](https://nodejs.org/download/release/v16.18.0/)

**Comment by TatyOko28** [comment]
The error indicates that your Node.js version (12.22.9) is too old for the Solana Jupiter bot, which requires Node.js >=16.
Solution: Update Node.js to a version >=16 and remove conflicting files like package-lock.json.
